File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int ptr = 0;
        for (int i = 0; i < nums.length-1; i++) {
            if (nums[i] != nums[i+1]) {
                nums[ptr++] = nums[i];
            }
        }

        nums[ptr] = nums[nums.length-1];
        return ptr+1;
    }
}

File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    ptr := 1
    for i := 1; i < len(nums); i++ {
        if (nums[i-1] != nums[i]) {
            nums[ptr] = nums[i]
            ptr++
        }
    }
    
    return ptr
}